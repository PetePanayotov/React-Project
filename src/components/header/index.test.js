import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';
import UserContext from '../../Context';
import {BrowserRouter , Route} from 'react-router-dom';

describe('Header Component' , () => {

    it('should render not authenticated routes' , () => {
        const tree = renderer.create(

            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        isLoggedIn: false,
                        user: {

                        }
                    }}>
                        <Header/>
                    </UserContext.Provider>

                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('should render authenticated routes' , () => {

        const tree = renderer.create(

            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        isLoggedIn: true,
                        user: {
                            username: 'tigyra',
                            userId: '1234'
                        }
                    }}>
                        <Header/>
                    </UserContext.Provider>

                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('should render admin routes' , () => {

        const tree = renderer.create(

            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        isLoggedIn: true,
                        isAdmin: true,
                        user: {
                            username: 'tigyra',
                            userId: '1234'
                        }
                    }}>
                        <Header/>
                    </UserContext.Provider>

                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    });

});