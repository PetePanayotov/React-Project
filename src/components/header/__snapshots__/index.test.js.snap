// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render admin routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="logo.jpg"
  />
  <div
    className="linkContainer"
  >
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/profile?userId=1234"
        onClick={[Function]}
      >
        [tigyra]
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/catalog"
        onClick={[Function]}
      >
        Catalog
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/create"
        onClick={[Function]}
      >
        Create
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/contacts"
        onClick={[Function]}
      >
        Contacts
      </a>
    </div>
    <div
      className="header-button-div"
    >
      <button
        className="header-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`Header Component should render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="logo.jpg"
  />
  <div
    className="linkContainer"
  >
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/profile?userId=1234"
        onClick={[Function]}
      >
        [tigyra]
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/catalog"
        onClick={[Function]}
      >
        Catalog
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/contacts"
        onClick={[Function]}
      >
        Contacts
      </a>
    </div>
    <div
      className="header-button-div"
    >
      <button
        className="header-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;

exports[`Header Component should render not authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="logo.jpg"
  />
  <div
    className="linkContainer"
  >
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/contacts"
        onClick={[Function]}
      >
        Contacts
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
    <div
      className="header-list-item"
    >
      <a
        className="header-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</header>
`;
